dev-notes/minimal-os-for-fobnail.md

```diff
@@ -0,0 +1,569 @@
+  $ cd secure-kernel-loader
+  $ make
+  $ cp skl.bin /mnt/boot/
+  ```
+
+Setup Zephyr build environment. Instructions below are based on Zephyr
+[Getting Started Guide](https://docs.zephyrproject.org/latest/develop/getting_started/index.html).
+
+- Install required packages. Ubuntu has too old packages so we have to use
+  KitWare repo.
+
+  ```shell
+  $ wget https://apt.kitware.com/kitware-archive.sh
+  $ sudo bash kitware-archive.sh
+  $ sudo apt install --no-install-recommends git cmake ninja-build gperf \
+    ccache dfu-util device-tree-compiler wget \
+    python3-dev python3-pip python3-setuptools python3-tk python3-wheel xz-utils file \
+    make gcc gcc-multilib g++-multilib libsdl2-dev python3-venv
+  $ pip3 install --user -U west
+  ```
+
+- Obtain Zephyr toolchain
+
+  ```shell
+  $ wget https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v0.14.1/zephyr-sdk-0.14.1_linux-x86_64.tar.gz
+  $ tar xf zephyr-sdk-0.14.1_linux-x86_64.tar.gz -C ~/.local
+  $ rm zephyr-sdk-0.14.1_linux-x86_64.tar.gz
+  $ ~/.local/zephyr-sdk-0.14.1/setup.sh -c
+  ```
+
+- Fetch Zephyr source code.
+
+  ```shell
+  $ west init zephyr
+  $ cd zephyr
+  $ west update
+  ```
+
+- Install required packages, use virtual environment to avoid conflict with
+  user/system packages.
+
+  ```shell
+  $ python3 -m venv .env
+  $ pip3 install -r zephyr/scripts/requirements.txt
+  ```
+
+- Build Zephyr, we use Qemu as the target, but it works on APU (without DLME).
+
+  ```shell
+  $ cd zephyr
+  $ west build -p auto -b qemu_x86_64 samples/userspace/hello_world_user/
+  $ sudo cp build/zephyr/zephyr.elf /mnt/boot/
+  ```
+
+- Configure GRUB, create '/mnt/boot/grub/grub.cfg` with the following contents.
+
+  ```shell
+  default 0
+  menuentry "Zephyr (DLME)" {
+    slaunch skinit
+    slaunch_module /skl.bin
+    multiboot /zephyr.elf
+  }
+  menuentry "Zephyr (no DLME)" {
+    multiboot /zephyr.elf
+  }
+  ```
+
+- Umount disk and sync before unplugging it. There were some problems with
+  obtaining working binary, so before booting Zephyr on target platform please
+  check whether it boots in Qemu (without DLME). When booting Zephyr you should
+  see `Hello World` message on serial console, if Zephyr doesn't output anything
+  or if it stuck at `Booting Zephyr` message then something went wrong.
+
+  ```shell
+  $ qemu-system-x86_64 -nographic -serial mon:stdio -m 512M -cpu qemu64 \
+    -drive format=raw,file=/dev/disk/by-id/usb-TS-RDF5_SD_Transcend_000000000039-0\:0
+  ```
+
+- Zephyr should normally boot on APU. If DLME variant (from boot menu) is used
+  Zephyr will also boot normally. That is because TrenchBoot GRUB executes SKL
+  only when using
+  [Multiboot2](https://github.com/3mdeb/grub/blob/tb_xen/grub-core/loader/multiboot.c#L192)
+  but Zephyr has support only for Multiboot1.
+
+- Booting Zephyr in DLME would require updating `slaunch` support in GRUB and
+  bringing Multiboot1 support to SKL.
```

Comment:
@arturkow2000 please either remove this or change it to adding MB2 to Zephyr.

Reply:
Done